tests/frontend/test_devicespecs.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Unit tests for the DeviceSpec class"""
+import textwrap
+
+import pytest
+
+import strawberryfields.devicespecs as ds
+from strawberryfields.devicespecs.device_specs import DeviceSpecs
+
+
+pytestmark = pytest.mark.frontend
+
+
+class TestAbstractDeviceSpec:
+    """Tests for the abstract DeviceSpec class.
+    This class primarily consists of abstract methods,
+    so only the methods containing logic are tested.
+    """
+
+    def test_no_parameter_ranges(self):
+        """If not defined, the parameter_ranges property
+        should return an empty dictionary."""
+
+        class DummyDevice(DeviceSpecs):
+            """Dummy device used to instantiate
+            the abstract base class"""
+            modes = 0
+            remote = False
+            interactive = True
+            primitives = set()
+            decompositions = set()
+
+        dummy = DummyDevice()
+
+        assert not dummy.parameter_ranges
```

Comment:
Per docstring, do we also want to test that it is a dictionary that is returned?

Reply:
Done